File: JavascriptAlgorithm/Map/twoSum-001.js
var twoSum = function (nums, target) {
  let map = new Map();
  for (let index = 0; index < nums.length; index++) {
    // 2:0 7:1
    if (map.get(target - nums[index]) !== undefined) {
      return [map.get(target - nums[index]), index];
    } else {
      map.set(nums[index], index);
    }
  }
  return [];
};

export { twoSum };

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SecondRound/Chapter05/twoSum_001.java
package algo.tzashinorpu.SecondRound.Chapter05;

import java.util.HashMap;

public class twoSum_001 {
    public int[] twoSum(int[] nums, int target) {
       /* 输入：nums = [2,7,11,15], target = 9
        输出：[0,1]*/
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                return new int[]{map.get(target - nums[i]), i};
            } else {
                map.put(nums[i], i);
            }
        }
        return new int[]{};
    }
}
